commands/http/multifilereader.go

```diff
@@ -64,13 +64,23 @@ func (mfr *MultiFileReader) Read(buf []byte) (written int, err error) {
 
 		// handle starting a new file part
 		if !mfr.closed {
-			if file.IsDirectory() {
+
+			var contentType string
+			if s, ok := file.(*files.Symlink); ok {
+				mfr.currentFile = s
+
+				// TODO(why): this is a hack. pick a real contentType
+				contentType = "symlink"
```

Comment:
(decoupled from this PR) Should the 'ipfs add' api be simplified where files are always archived (and optionally gzipped)?
This is already what `ipfs get` does.

Will this simplify api writing (e.g. then multipart handling will just be archive.Extract() in each lang)?